techmatters/hrm
# [WIP] Exposes permissions API

## Description
This PR
- Moves most files related to permissions framework to TS.
- Factors out the rules map (`{ [key: string]: RulesFile }`) to consume the rules definitions files while being imported from a single place.
- Adds a `/permissions` endpoint that will retrieve the permissions rules for the given account (if they are defined).

WIP:
- Tests

### Checklist

### Verification steps

- Check that all the backend side permissions are still in place.
- Start local server with a valid `STATIC_KEY_<accountSid>=<some permissions file key>`.
- Perform a GET on 'http://localhost:3000/v0/accounts/<accountSid>/permissions' -H "Authorization: Bearer <valid token>" to retrieve the rules corresponding to `<some permissions file key>`.
- Change `STATIC_KEY_<accountSid>` value to some invalid key and hit again to reach error.
